### PR TITLE
deps(noise): update `x25519-dalek` to 2.0.0

### PR DESCRIPTION
**Pull request by kayabaNerve**
## Notes & open questions

Updates libp2p-noise's x25519-dalek dependency to 2.0.0.

Additionally bumps the version to:

1) Provide a location in the CHANGELOG for the changes.
2) Encourage a new release with this, since it doesn't appear to be used in any public API and therefore isn't a breaking change. Solely a benefit to performance and a reduction in dependency tree size.

Apologies if this was over-eager, and I'm happy to open a PR of solely the first commit.

I will note I'm unaware why this wasn't done earlier. Possibly an oversight? Or I'm missing something.

I did not add tests, yet I did run the existing test suite.

## Change checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: transports/noise/Cargo.toml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name = "libp2p-noise"
 edition = "2021"
 rust-version = { workspace = true }
 description = "Cryptographic handshake protocol using the noise framework."
-version = "0.43.1"
+version = "0.43.2"
```
Comment (thomaseizinger) [inline transports/noise/Cargo.toml:6]: You'll also have to adjust the version in the root manifest under `workspace.dependencies`.

**Review by thomaseizinger — APPROVED** [review]
Thanks!

Yeah happy to release right away once merged. Just one comment!

**Comment by thomaseizinger** [comment]
> I'm unsure why this wasn't done prior since it doesn't seem exposed in any public API.

We use dependabot and it seems I accidentally closed that one: https://github.com/libp2p/rust-libp2p/pull/4355.

**Comment by kayabaNerve** [comment]
Updated there too :+1:

**Comment by thomaseizinger** [comment]
Released!

**Pull request merged**